### PR TITLE
Update Referential Integrity metric to support NaNs in child column 

### LINKED ISSUE #587: Update Referential Integrity metric to support NaNs in child column
### Problem Description
- For the Referential Integrity metric, we currently do the following:
  - If the foreign key in the real data has any NaN values, those are dropped in the foreign key in the synthetic data.
  - We also verify that all foreign keys have valid parent.
- To support NaNs values in foreign keys, we need to update this logic. 

### Expected behavior
- Update `ReferentialIntegrity.compute_breakdown()` to do the following:
  - If the real data has NaN values, and the synthetic data has NaN values in the FK, then it is okay.
  - If the real data does not have NaN values, but the synthetic data has NaN values in the FK, this will result in a lower score. 
    - The score can be lowered by the number of extra NaNs present in the synthetic data.
- Update test_referential_interity.py
- Update documentation

### Additional context
- Function to update:
https://github.com/sdv-dev/SDMetrics/blob/51f3ac2904c065462097c80f4387b973657ad6d9/sdmetrics/column_pairs/statistical/referential_integrity.py#L36

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86b0uyzmn
Resolve #587

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b0uyzmn SDMetrics - Update ReferentialIntegrity to support NaNs in child column  #587](https://app.clickup.com/t/86b0uyzmn)

File: tests/unit/column_pairs/statistical/test_referential_integrity.py [hunk 1]
```diff
@@ -68,32 +68,56 @@ def test_compute(self, compute_breakdown_mock):
...
+        )
+        result_2 = metric.compute(
+            real_data=(parent_keys, real_fk), synthetic_data=(parent_keys, synth_fk_2_nan)
         )
 
         # Assert
-        assert result == 0.8
+        assert result_0 == 1.0
+        assert result_1 == 0.8
+        assert result_2 == 0.6666666666666666
```
Comment (gsheni) [inline tests/unit/column_pairs/statistical/test_referential_integrity.py:123]:
```suggestion
        assert result_2 == (2/3)
```

**Review by rwedge — APPROVED** [review]
Looks good.

**Review by rwedge — COMMENTED** [review]
~~Can we look at the case where the real data has a NaN in the foreign kyes but the synth does not?~~ Already present

**Review by gsheni — APPROVED** [review]

**Pull request merged**